src/ProgressOnderwijsUtils/Win32/VirusScan.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using Microsoft.Windows.Sdk;
+
+namespace ProgressOnderwijsUtils.Win32
+{
+    public static class VirusScan
+    {
+        public static unsafe bool IsMalware(byte[] buffer, string contentName, string sessionName)
+        {
+            PInvoke.AmsiInitialize(sessionName, out var context).AssertResultOk();

```

Comment:
Ik denk dat AssertResultOk beter in de try kan?

Reply:
dat kan wel, maar dan moet ik context apart delcareren voor de try

Reply:
Nu roep je uninitialize niet aan als AssertResultOk crasht. (Niet dat ik zeker weet of dat kwaad kan of überhaupt mag.)